styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -62,4 +72,46 @@ public FlyteExecution createExecution(final String name, final FlyteExecConf fly
       throw new CreateExecutionException(flyteExecConf, e);
     }
   }
+
+  public void poll(final String project, final String domain, final String name, RunState runState) {
+    Objects.requireNonNull(project);
+    Objects.requireNonNull(domain);
+    Objects.requireNonNull(name);
+    final ExecutionOuterClass.Execution execution =
+        flyteAdminClient.getExecution(project, domain, name);
+    emitFlyteEvents(execution, runState);
+  }
+
+  private void emitFlyteEvents(ExecutionOuterClass.Execution execution, RunState runState) {
+    final Execution.WorkflowExecution.Phase phase = execution.getClosure().getPhase();
+    final FlytePhase flytePhase = FlytePhase.fromProto(phase);
+    try {
+      switch (flytePhase) {
+        case SUCCEEDED:
+          stateManager.receive(Event.terminate(runState.workflowInstance(), Optional.of(0)));
+          break;
+        case FAILED:
+        case ABORTED:
+        case TIMED_OUT:
+          final String flyteCode = execution.getClosure().getError().getCode();
+          final int styxCode = flyteCodeToStyx(flyteCode);
+          stateManager.receive(Event.terminate(runState.workflowInstance(), Optional.of(styxCode)));
+          break;
+      }
+    } catch (IsClosedException e) {
+      return;
```

Comment:
Please log a warning here as it is done in the Handlers

Reply:
The equivalent check is ignored in the Kuberenetes handler here https://github.com/spotify/styx/blob/master/styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java#L663

Reply:
Should I add it anyway?

Reply:
DockerRunnerHandler and FlyteRunnerHandlers both logs on catching `IsClosedException`. I see that this have been done inconsistently but I sill prefer to err on the side of more login